### ISSUE TITLE
subscriptions: merged query is not used

### ISSUE THREAD

**Issue by oliver-sanders** [cylc/cylc-ui#1110]
Discovered whilst trying to understand why I can't get the Tree and Graph views running at the same time.

When you open a workflow in the UI it creates a subscription. When you open another "view" on the workflow the subscription gets stopped, the new subscription is merged into it and the new subscription is re-issued.

However, it doesn't look like the new query is actually being picked up, you just get whatever subscription was issued first.

I jammed in a breakpoint into the UIS here:

https://github.com/cylc/cylc-uiserver/blob/549afbd99389a5e29571b3f5f71cc6e4505a4270/cylc/uiserver/handlers.py#L388-L389

And inspected the websocket messages like so:

```python
(Pdb) import json
(Pdb) msg = json.loads(message)
(Pdb) print(msg['payload']['query'])
```

To inspect the query, which confirmed that the updated query was not making it through to the UIS.

This means that the subscription merging isn't actually hooked into anything at the moment. The Tree & Table views are able to co-exist only because their queries overlap.

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

**Comment by oliver-sanders** [comment]
Figured it, the query object contains derived properties. These properties are not updated when we modify the object, `ApolloClient` appears to use these properties to determine the subscription.

The solution is to re-create the query object when the subscription changes rather than to modify it in place.
